### PR TITLE
fix: banner alert to render multiple general alerts

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**
This PR aims to fix the banner alert to support rendering multiple alerts. Previously we only rendered one alert and if there were more alerts we rendered the banner with a default copy informing the user there are multiple alerts.

- Fixed padding on the alerts modal based on [figma](https://www.figma.com/design/gcwF9smHsgvFWQK83lT5UU/Confirmation-redesign-V4?node-id=3355-12480&node-type=frame&t=3Vbe0qFcmcfN5uCG-0)
- Fixed bug Contract Interaction and Alerts - 'Cannot read properties of undefined (reading 'key')`

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27339?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/2873 https://github.com/MetaMask/metamask-extension/issues/27238

## **Manual testing steps**

1. Create a transaction with high nonce
2. Go to test dapp
3. Trigger a malicious transaction from PPOM session

### **After**

Screenshot from 2024-09-23 13-38-10

Screenshot from 2024-09-23 14-09-42

Screenshot from 2024-09-23 14-21-53

[bug.webm](https://github.com/user-attachments/assets/eb447959-78f0-4ccc-a554-cca272e59b19)

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/app/alert-system/multiple-alert-modal/multiple-alert-modal.tsx [hunk 1]
```diff
@@ -162,7 +162,11 @@ export function MultipleAlertModal({
     initialAlertIndex === -1 ? 0 : initialAlertIndex,
   );
 
-  const selectedAlert = alerts[selectedIndex];
+  // If the selected alert is not found, default to the first alert
+  const selectedAlert = alerts[selectedIndex]
+    ? alerts[selectedIndex]
+    : alerts[0];
```
Comment (digiwand) [inline ui/components/app/alert-system/multiple-alert-modal/multiple-alert-modal.tsx:168]: [nit] we can shorten this to 
```suggestion
  const selectedAlert = alerts[selectedIndex] ?? alerts[0];
```

**Review by digiwand — DISMISSED** [review]
Lgtm!

**Review by digiwand — DISMISSED** [review]

**Review by jpuri — DISMISSED** [review]

File: ui/pages/confirmations/components/confirm/title/title.tsx [hunk 2]
```diff
@@ -25,37 +24,27 @@ import { getIsRevokeSetApprovalForAll } from '../info/utils';
...
-            ? t('alertBannerMultipleAlertsDescription')
-            : singleAlert.message
-        }
-        severity={highestSeverity}
-        provider={hasMultipleAlerts ? undefined : singleAlert.provider}
-        details={hasMultipleAlerts ? undefined : singleAlert.alertDetails}
-        reportUrl={singleAlert.reportUrl}
-      />
+    <Box marginTop={3}>
+      {generalAlerts.map((alert) => (
```
Comment (matthewwalsh0) [inline ui/pages/confirmations/components/confirm/title/title.tsx:35]: Are we concerned about the order? Highest severity first?

**Review by matthewwalsh0 — DISMISSED** [review]

File: ui/pages/confirmations/components/confirm/title/title.tsx (hunk 2 above)
Comment (vinistevam) [reply ui/pages/confirmations/components/confirm/title/title.tsx:35]: Good point, as we are ordering modals and inline alerts. I will ask @SayaGT.

Comment (vinistevam) [reply ui/pages/confirmations/components/confirm/title/title.tsx:35]: I confirmed with Saya, that the highest severity should be first. Done [here](https://github.com/MetaMask/metamask-extension/pull/27339/commits/9fbc80d42ad5369382a2b8b35669b7239360accf)

File: ui/hooks/useAlerts.test.ts [hunk 3]
```diff
@@ -156,16 +163,42 @@ describe('useAlerts', () => {
...
+      const result = renderAndReturnResult(undefined, {
+        confirmAlerts: {
+          alerts: {
+            [ownerIdMock]: expectedGeneralAlerts,
+          },
+          confirmed: {},
+        },
+      });
+      expect(result.current.generalAlerts).toEqual(
+        sortAlertsBySeverity(expectedGeneralAlerts),
```
Comment (matthewwalsh0) [inline ui/hooks/useAlerts.test.ts:200]: If this is a requirement of the logic, should we instead test the order directly, rather than exporting code that is private to the hook?

**Review by matthewwalsh0 — CHANGES_REQUESTED** [review]

File: ui/hooks/useAlerts.test.ts (hunk 3 above)
Comment (vinistevam) [reply ui/hooks/useAlerts.test.ts]: Right, set the `sortAlertsBySeverity` to private and tested the logic directly [here](https://github.com/MetaMask/metamask-extension/pull/27339/commits/351bbed65cca27291a9f9515f1ca812813835db7).

**Comment by sleepytanya** [comment]
BigNumber error on Base for 'Malicious approval', other PPOM transactions are slow, error for Malicious ETH transfer  ```
TypeError: Cannot set properties of undefined (setting'gasFeeEstimates')
```:


https://github.com/user-attachments/assets/10891e7c-b4cc-4128-8c0d-656a83fffe60

**Comment by sleepytanya** [comment]
Multiple alerts work as expected

https://github.com/user-attachments/assets/4b47bca3-95ab-4195-9475-b634b5776889

**Comment by sleepytanya** [comment]
Multiple alerts work as expected:


https://github.com/user-attachments/assets/a7ec1ba1-d49f-41c6-8a54-d233091ffb6e



Extension breaks with BigNumber error when trying to send `Malicious ERC20 Approval` on all popular networks except Linea mainnet and zkSync. `Malicious ERC20 Approval` works on develop and RC 12.4.0.


Error on current branch:


Screenshot 2024-10-04 at 00 38 11


https://github.com/user-attachments/assets/6421e085-cec1-462e-a4b0-d7387b13295e


Before it breaks you can see two screens - `Allowance request `


Screenshot 2024-10-04 at 00 34 21


and then `Remove Permission`:



Screenshot 2024-10-04 at 00 34 36

**Review by matthewwalsh0 — APPROVED** [review]

**Review by OGPoyraz — APPROVED** [review]
